From 52316a94725578b18875dbc848ff049844d105f2 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Sun, 17 Sep 2017 12:28:17 +0300
Subject: [PATCH 20/35] Fix AAC and AMR format recording hangs When using AAC
 and AMR encoder, stops recording during pause state would block by mSource

Change-Id: Ib81bca29e5e634e8152fe7604d8662ad0909d206
---
 media/libstagefright/AACWriter.cpp | 10 ++--------
 media/libstagefright/AMRWriter.cpp | 10 ++--------
 2 files changed, 4 insertions(+), 16 deletions(-)

diff --git a/media/libstagefright/AACWriter.cpp b/media/libstagefright/AACWriter.cpp
index 8b1e1c3..98bdea0 100644
--- a/media/libstagefright/AACWriter.cpp
+++ b/media/libstagefright/AACWriter.cpp
@@ -153,17 +153,11 @@ status_t AACWriter::reset() {
 
     mDone = true;
 
+    mSource->stop();
+
     void *dummy;
     pthread_join(mThread, &dummy);
-
     status_t err = static_cast<status_t>(reinterpret_cast<uintptr_t>(dummy));
-    {
-        status_t status = mSource->stop();
-        if (err == OK &&
-            (status != OK && status != ERROR_END_OF_STREAM)) {
-            err = status;
-        }
-    }
 
     mStarted = false;
     return err;
diff --git a/media/libstagefright/AMRWriter.cpp b/media/libstagefright/AMRWriter.cpp
index 961b57f..0fd626e 100644
--- a/media/libstagefright/AMRWriter.cpp
+++ b/media/libstagefright/AMRWriter.cpp
@@ -148,17 +148,11 @@ status_t AMRWriter::reset() {
 
     mDone = true;
 
+    mSource->stop();
+
     void *dummy;
     pthread_join(mThread, &dummy);
-
     status_t err = static_cast<status_t>(reinterpret_cast<uintptr_t>(dummy));
-    {
-        status_t status = mSource->stop();
-        if (err == OK &&
-            (status != OK && status != ERROR_END_OF_STREAM)) {
-            err = status;
-        }
-    }
 
     mStarted = false;
     return err;
-- 
2.7.4

